client/src/pages/Invoice/InvoiceList.jsx

```diff
@@ -1,21 +1,194 @@
+        if (customer.Name == name) {
+          return customer.CustomerID;
+        }
+      }
+    }
+  };
+
+  const navigate = useNavigate();
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    const representativeId = getCustomerIdByName(e.target.representative.value);
+    console.log(representativeId);
+    const bookings = selectedBookingList.map((booking) => booking.bookingId);
+
+    if (bookings.length === 0) {
+      alert('Please select at least 1 booking');
+      return;
+    }
+
+    const data = {
+      RepresentativeId: representativeId,
+      Bookings: bookings,
+    };
+
+    const response = await createInvoice(data);
+
+    if (response.status === 201) {
+      const invoiceId = await response.json();
+
+      navigate(`/invoices/${invoiceId}`);
+    } else {
+      alert('Failed to create invoice');
+      console.error(response);
+    }
+  };
+
+  return (
+    <div className="flex flex-col w-full py-4 px-2 min-h-80">
+      <Title title="Invoice List" />
+
+      <div className="lg:w-[70%] w-full mx-auto overflow-x-auto">
+        <table className="table-auto text-center w-full">
+          <thead className="table-header-group md:text-xl text-lg font-play text-zinc-100">
+            <tr className="table-row">
+              <th className="border bg-orange md:h-12 h-10 px-2">No</th>
+              <th className="border bg-orange px-2">Room</th>
+              <th className="border bg-orange px-2">Booking Date</th>
+              <th className="border bg-orange px-2">Nights</th>
+              <th className="border bg-orange px-2">Price</th>
+              <th className="border bg-orange px-2">Select</th>
+            </tr>
+          </thead>
+          <tbody className="md:text-lg text-base font-amethysta">
+            {invoices.map((invoice, index) => (
+              <tr key={index}>
+                <td className="border py-2 border-gray">{index + 1}</td>
+                <td className="border px-2 border-gray">
+                  {invoice.RoomNumber}
+                </td>
+                <td className="border px-2 border-gray">
+                  {formatDate(invoice.BookingDate)}
+                </td>
+                <td className="border px-2 border-gray">{invoice.Nights}</td>
+                <td className="border px-2 border-gray">
+                  {'$' + invoice.Price}
+                </td>
+                <td className="border px-2 border-gray">
+                  <input
+                    type="checkbox"
+                    onClick={handleSelect}
+                    value={index}
+                    className="cursor-pointer w-5 h-5 appearance-none border border-gray rounded-md checked:bg-orange checked:border-transparent mt-1"
+                  />
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </table>
+
+        <form
+          className="flex justify-between mt-6 gap-2"
+          onSubmit={handleSubmit}
```

Comment:
- Oh no. This is terrible because it means that the CREATE function cannot be executed in any case
- Unfortunately, this does not happen on my machine. I can still create invoices normally so I don't know what the problem is :<
- It would be great if you could send me more details by: Ctrl + Shift + I -> Network -> Check the response returned when calling API POST /api/invoices

- This is the process on my machine. You can refer to it to know how to get response.
BEFORE:
![image](https://github.com/user-attachments/assets/0dd4f086-1643-4148-81a5-083c5eb8c7c2)

AFTER:
![image](https://github.com/user-attachments/assets/f5beb9d3-2fcd-44f5-80c2-82a7cb2cceed)

Reply:
I will check this out ASAP

Reply:
**It seems that the issue has something to do with the BE?**
Based on @eNKay2408 's suggestion, I found some issues related to the invoice creation functionality

When hitting 'CREATE INVOICE'
![image](https://github.com/user-attachments/assets/31d643d6-3b45-4eda-8a2a-6fe4798f1252)

After pressing 'OK'
![image](https://github.com/user-attachments/assets/5125fc2b-c091-4e67-8d08-115e2e241259)

_Is it possible that this has something to do with backend side? Potentially with setting CORS access policy?_

Also, when I first load Invoice page, it seems that the Booking with id = 0 is not loaded
![image](https://github.com/user-attachments/assets/76fca986-cca6-4801-99ce-2b741a7574c5)

_Choosing the very first row in the list, the id responded is 1_
![image](https://github.com/user-attachments/assets/edbc55e1-ae45-4399-bcf8-92cfdf9fad9a)

Reply:
there are no booking that have id =  0
![image](https://github.com/user-attachments/assets/0081da5a-1d54-4d01-9f4e-7b3afe0e1999)
Post: http://localhost:3000/api/invoices with 2 booking in first image still work
![image](https://github.com/user-attachments/assets/4285d8aa-0bf6-4074-b82d-cecd37d32675)